feature/qos/otg_tests/ingress_police_nhg/README.md

```diff
@@ -0,0 +1,363 @@
+
+
+### TE-18.2.3 Test traffic
+
+* Send traffic
+  * Send traffic from ATE port 1 to DUT for dest_A and is conforming to cir.
+  * Send traffic from ATE port 1 to DUT for to dest_B and is conforming to
+    cir.
+  * Validate packets are received by ATE port 2.
+    * Validate qos interface scheduler counters
+    * Validate afts next hop counters
+  * Validate outer packet ipv6 flow label assignment
+    * When the outer packet is IPv6, the flow-label should be inspected on the ATE.
+      * If the inner packet is IPv4, the outer IPv6 flow label should be computed based on the IPv4 5 tuple src,dst address and ports, plus protocol
+      * If the inner packet is IPv6, the inner flow label should be copied to the outer packet.
+  * Increase traffic on flow to dest_B to 2Gbps
+    * Validate that flow dest_B experiences ~50% packet loss (+/- 1%)
+
+#### OpenConfig Path and RPC Coverage
+
+```yaml
+paths:
+  # qos scheduler config
+  /qos/scheduler-policies/scheduler-policy/config/name:
+  /qos/scheduler-policies/scheduler-policy/schedulers/scheduler/config/type:
+  /qos/scheduler-policies/scheduler-policy/schedulers/scheduler/one-rate-two-color/config/cir:
+  /qos/scheduler-policies/scheduler-policy/schedulers/scheduler/one-rate-two-color/config/bc:
+  /qos/scheduler-policies/scheduler-policy/schedulers/scheduler/one-rate-two-color/config/queuing-behavior:
+  /qos/scheduler-policies/scheduler-policy/schedulers/scheduler/one-rate-two-color/exceed-action/config/drop:
+
+  # qos classifier config
+  /qos/classifiers/classifier/config/name:
+  /qos/classifiers/classifier/terms/term/config/id:
+  #/qos/classifiers/classifier/terms/term/conditions/next-hop-group/config/name: # TODO: new OC leaf to be added
+
+  # qos policer config - TODO: a new OC subtree (/qos/policer-policies, essentially copying/moving policer action from schedulers)
+  # /qos/policer-policies/policer-policy/config/name:
+  # /qos/policer-policies/policer-policy/config/policers/policer/config/sequence:
+  # /qos/policer-policies/policer-policy/config/policers/policer/one-rate-two-color/config/cir:
+  # /qos/policer-policies/policer-policy/config/policers/policer/one-rate-two-color/config/bc:
+  # /qos/policer-policies/policer-policy/config/policers/policer/one-rate-two-color/config/cir:
+  # /qos/policer-policies/policer-policy/config/policers/policer/one-rate-two-color/exceed-action/config/drop:
+
+  # qos interface scheduler counters
+  /qos/interfaces/interface/input/scheduler-policy/schedulers/scheduler/state/conforming-pkts:
+  /qos/interfaces/interface/input/scheduler-policy/schedulers/scheduler/state/conforming-octets:
+  /qos/interfaces/interface/input/scheduler-policy/schedulers/scheduler/state/exceeding-pkts:
+  /qos/interfaces/interface/input/scheduler-policy/schedulers/scheduler/state/exceeding-octets:
+
+rpcs:
+  gnmi:
+    gNMI.Set:
+      union_replace: true
+      replace: true
+    gNMI.Subscribe:
+      on_change: true
+```
+
+## Required DUT platform
+
+* FFF
```

Comment:
Can this not be tested on a virtualised platform?

Reply:
Since this test includes dataplane requirements , it must also be testing on the real target platform.  
If only run on a virtual platform, that would certify the virtualized platform is compliant.  But it would not certify a platform implementing the policing on an ASIC is compliant.  In our use case, we want to know the hardware is working.  (when developing the code for this test, I would absolutely try and use a virtualized platform until I got the test perfected, then run on hardware).

TE-18.1 focuses only on gRIBI aft manipulation and verifies via packet forwarding.  A control-plane only version of that test could be written (which doesn't test packet forwarding), and run on a virtualized platform.  But a test with packet forwarding in the dataplane test needs to run on the real target platform.